### DISCUSSION TITLE
How do I get layer data in my API call

### DISCUSSION THREAD

**Discussion by dkanze** [weather-gov/api#251] [Q&A]
I am new to API use and have been experimenting https://api.weather.gov/gridpoints/LWX/96,70/forecast and https://api.weather.gov/gridpoints/LWX/96,70/forecast/hourly.  I am able to return the data offered for various office/gridpoints.  I have seen on https://weather-gov.github.io/api/gridpoints that layer data not included in the daily/hourly API calls above.  My question is, how do I request the additional layer data to be returned in my calls?

Thanks
Dan

**Comment by JJKraw** [comment]
Try https://api.weather.gov/gridpoints/LWX/96,70

**Comment by dkanze** [comment]
This looks like it will keep me busy for a while.  Is there documentation to interpreting some of the values?

Thank you very much.
Dan

**Comment by JJKraw** [comment]
I know of nothing beyond the page you posted.

**Comment by dkanze** [comment]
After looking at what is returned, I am guessing that each of the items (temperature, dewpoint, windSpeed, etc.) are to be considered individually and not laid side by side.  I say this because the timestamps do not align.  Do I have this right?

Thanks
Dan

**Comment by dkanze** [comment]
Ok I see now.
The layer data is presented as a time series.
Much of the data is calculated for each hour of the forecast period, but to conserve bandwidth the API will merge consecutive values that are equal.
Each data point will have the following properties:
validTime: This is the time interval that the value applies to. The interval is represented in ISO 8601 format.
For example, 2019-07-04T18:00:00+00:00/PT3H represents an interval starting on July 4, 2019 at 1800 UTC and continuing for 3 hours (from 1800 to 2100).

So to get items to align, I will need to repeat some rows by the number of hours indicated after timestamp.  3H in the example above.

Thank you again for your assistance.
Dan

**Comment by JJKraw** [comment]
Note that if you are dealing with items like rain or snow accumulation, you need to deal with that amount accumulating over the entire interval. E.g., if it says 10 inches of snowfall over 8 hours, and you are trying to use that in per-hour pieces, you need to divvy it up (2.5 inches for each hour).
